www/class/centreonAuth.SSO.class.php

```diff
@@ -194,91 +224,126 @@ protected function checkSsoClient()
-     * @param string $clientId Keycloak Client ID
-     * @param string $clientSecret Keycloak Client Secret
-     * @param string $code Keycloak Authorization Code
+     * @param string $url          OpenId Connect Client Token endpoint
+     * @param string $redirectUri  OpenId Connect Redirect Url
+     * @param string $clientId     OpenId Connect Client ID
+     * @param string $clientSecret OpenId Connect Client Secret
+     * @param string $code         OpenId Connect Authorization Code
+     * @param bool   $verifyPeer   disable SSL verify peer
      *
      * @return string
     */
-    public function getKeycloakToken($base, $realm, $redirectUri, $clientId, $clientSecret, $code)
+    public function getOpenIdConnectToken($url, $redirectUri, $clientId, $clientSecret, $code, $verifyPeer)
     {
-        $url = $base . "/realms/" . $realm . "/protocol/openid-connect/token";
-        $data = array(
+        $data = [
             "client_id" => $clientId,
             "client_secret" => $clientSecret,
             "grant_type" => "authorization_code",
             "code" => $code,
             "redirect_uri" => $redirectUri
-        );
+        ];
+
         $ch = curl_init($url);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
         curl_setopt($ch, CURLOPT_POST, true);
         curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($data));
 
+        if ($verifyPeer) {
+            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        }
+
         $result = curl_exec($ch);
         curl_close($ch);
 
         $resp = json_decode($result, true);
 
-        return $resp["access_token"];
+        return $resp["access_token"] ?? null;
     }
 
     /**
-     * Validate Centreon user on Keycloak
+     * Validate Token on OpenId Connect
      *
-     * @param string $base Keycloak Server Url
-     * @param string $realm Keycloak Client Realm
-     * @param string $clientId Keycloak Client ID
-     * @param string $clientSecret Keycloak Client Secret
-     * @param string $token Keycloak Token Access
+     * @param string $url          OpenId Connect Introspection Token Endpoint
+     * @param string $clientId     OpenId Connect Client ID
+     * @param string $clientSecret OpenId Connect Client Secret
+     * @param string $token        OpenId Connect Token Access
+     * @param bool   $verifyPeer   disable SSL verify peer
      *
-     * @return string
+     * @return array
      */
-    public function getKeycloakUserInfo($base, $realm, $clientId, $clientSecret, $token)
+    public function getOpenIdConnectIntrospectionToken($url, $clientId, $clientSecret, $token, $verifyPeer)
     {
-        $url = $base . "/realms/" . $realm . "/protocol/openid-connect/token/introspect";
-        $data = array(
+        $data = [
             "token" => $token,
             "client_id" => $clientId,
             "client_secret" => $clientSecret
-        );
+        ];
 
         $ch = curl_init($url);
```

Comment:
same

Reply:
cURL does not throw exceptions, it is a thin wrapper for the C library it binds to.